Clustering/Hdbscan/src/main/java/org/tribuo/clustering/hdbscan/HdbscanModel.java

```diff
@@ -91,6 +95,12 @@
 
     @Override
     public Prediction<ClusterID> predict(Example<ClusterID> example) {
+        // A deserialized model which doesn't have distType set needs to handle this case. This check can be removed
+        // in a major version release when the deprecated field no longer needs to be supported.
+        DistanceType distanceType = distType;
```

Comment:
I think it would be better to bounce distanceType into distType in the deserialization readObject method. You can see multiple places where we use this trick to upgrade older models (e.g. the regression models to fix the indices bug, or the linear sgd models when they were retrofitted with a superclass).